lib/server.js

```diff
@@ -34,48 +34,52 @@ const CUSTOM_PLUGINS = [
  * @return {http.Server}                            A listener: NodeJS http.Server object
  */
 module.exports = async (config) => {
-    let corsOrigins = [config.ecosystem.ui];
+    try {
+        let corsOrigins = [config.ecosystem.ui];
 
-    if (Array.isArray(config.ecosystem.allowCors)) {
-        corsOrigins = corsOrigins.concat(config.ecosystem.allowCors);
-    }
+        if (Array.isArray(config.ecosystem.allowCors)) {
+            corsOrigins = corsOrigins.concat(config.ecosystem.allowCors);
+        }
 
-    // Allow ui to query store
-    const cors = {
-        origin: corsOrigins
-    };
+        // Allow ui to query store
+        const cors = {
+            origin: corsOrigins
+        };
 
-    // Create a server with a host and port
-    // Use Object.assign to pass httpd config
-    const server = Hapi.server({
-        routes: {
-            cors,
-            log: { collect: true },
-            payload: { timeout: 60000 }
-        },
-        router: {
-            stripTrailingSlash: true
-        },
-        cache: config.cache,
-        ...config.httpd
-    });
+        // Create a server with a host and port
+        // Use Object.assign to pass httpd config
+        const server = Hapi.server({
+            routes: {
+                cors,
+                log: { collect: true },
+                payload: { timeout: 60000 }
+            },
+            router: {
+                stripTrailingSlash: true
+            },
+            cache: [config.cache],
+            ...config.httpd
+        });
 
-    // Register Plugins
-    /* eslint-disable */
-    const coreRegistrations = CORE_PLUGINS.map(plugin => require(plugin));
-    const customRegistrations = CUSTOM_PLUGINS.map(plugin => ({
-        plugin: require(`../plugins/${plugin}`),
-        options: config[plugin] || {}
-    }));
-    /* eslint-enable */
+        // Register Plugins
+        /* eslint-disable */
+        const coreRegistrations = CORE_PLUGINS.map(plugin => require(plugin));
+        const customRegistrations = CUSTOM_PLUGINS.map(plugin => ({
+            plugin: require(`../plugins/${plugin}`),
+            options: config[plugin] || {}
+        }));
+        /* eslint-enable */
 
-    await server.register(coreRegistrations.concat(customRegistrations), {
-        routes: {
-            prefix: '/v1'
-        }
-    });
+        await server.register(coreRegistrations.concat(customRegistrations), {
+            routes: {
+                prefix: '/v1'
+            }
+        });
 
-    await server.start();
+        await server.start();
 
-    return server;
+        return server;
+    } catch (err) {
+        throw err;
```

Comment:
`try catch block` would prevent pod from crashing. Do we need to log the errors, or k8s takes care of it?

Reply:
Error is thrown upward and logged.
Its here. https://github.com/screwdriver-cd/store/blob/master/bin/server#L42

Reply:
catch to throw is not needed, you can remove the try..catch and let caller handle the error.